#**Problem Set I: Mecánica Celeste:** El problema de los n cuerpos

<div align="left">

### Juliana Ruiz Montoya
### 1007435437

###**1. Plano invariable del sistema Sol-Júpiter-Saturno (Parte 1)**

Encuentre los vectores de estado del Sol, Júpiter y Saturno medidos respecto al baricentro del sistema solar (un origen inercial) para el día de su cumpleaños a las 00:00:00 de tiempo universal y con base en esto:

**(a)** Calcule el momento angular de cada cuerpo y el momento angular total del sistema, compare e interprete los resultados. NOTA: Tenga mucho cuidado con las unidades.

**(b)** Encuentre la inclinación del plano invariable de Laplace del sistema Sol-Júpiter-Saturno con respecto al plano de la eclíptica.

**(c)** Compare la ubicación de este plano invariable con la ubicación de los planos de las órbitas de Júpiter y Saturno si se sabe que la inclinación de sus órbitas (de nuevo medidas respecto a la eclíptica) son respectivamente 1◦18′ y 2◦29′.

**(d)** Repita los literales anteriores 100 años después de la fecha de su cumpleaños, compare e interprete los resultados.

NOTA: Recuerden que esta primera parte deberá ser realizada ”manualmente”, es decir, el código lo harán ustedes de cero.

In [1]:
import numpy as np
import pandas as pd

Se obtienen los vectores de estado los vectores de estado (vector posición $\vec{r}$ y el vector velocidad $\vec{v}$) para los cuerpos del Sol, Júpiter y Saturno para la fecha de 2024-10-31, a través de Horizons System - NASA (https://ssd.jpl.nasa.gov/horizons/app.html#/)

In [2]:
# Vectores de estado del Sol, Júpiter, Saturno

pos_sol = np.array([-9.223125953892386E+05, -6.971582947371479E+05, 2.789422119026093E+04]) * 1000 # en m
pos_jup = np.array([2.256086993617021E+08, 7.219617753824757E+08, -8.042224237805039E+06]) * 1000 # en m
pos_sat = np.array([1.406956569228385E+09, -3.152916834129006E+08, -5.053584979970349E+07]) * 1000 # en m

vel_sol = np.array([1.183203802762799E-02, -7.576359162400811E-03, -1.817523289842603E-04]) * 1000 # en m/s
vel_jup = np.array([-1.261884072114931E+01, 4.519624847356097E+00, 2.635982390958225E-01]) * 1000 # en m/s
vel_sat = np.array([1.575053643300134E+00, 9.405962361845964E+00, -2.262721192052810E-01]) * 1000 # en m/s

In [3]:
# Masas de cada planeta

m_sol = 1.98840987e+24 # en kg
m_jup = 1.89851767e+21 # en Kg
m_sat = 5.68457894e+20 # en Kg

Ahora se calcula el momento angular de cada cuerpo, el cual se define así:

$$\vec{L} = ∑_i m_i (\vec{r}_i × \vec{v}_i)$$

donde $m$ son las masas de cada cuerpo, $\vec{r}$ son los vectores posición y $\vec{v}$ son los vectores velocidad.

In [4]:
''' Momentum lineal de los planetas '''

# Momento angular
L_sol = np.cross(pos_sol, vel_sol) * m_sol
L_jup = np.cross(pos_jup, vel_jup) * m_jup
L_sat = np.cross(pos_sat, vel_sat) * m_sat

print("El momento angular del sol es", L_sol, "kg m^2 / s")
print("El momento angular de Júpiter es", L_jup, "kg m^2 / s")
print("El momento angular de Saturno es", L_sat, "kg m^2 / s")

El momento angular del sol es [6.72175558e+32 3.22943659e+32 3.02965560e+34] kg m^2 / s
El momento angular de Júpiter es [4.30309831e+35 7.97633547e+34 1.92319599e+37] kg m^2 / s
El momento angular de Saturno es [3.10764572e+35 1.35724080e+35 7.80514395e+36] kg m^2 / s


In [5]:
# Normalizar momento angular (magnitud escalar)
Lsol_norm = np.linalg.norm(L_sol)
Ljup_norm = np.linalg.norm(L_jup)
Lsat_norm = np.linalg.norm(L_sat)

# Momento angular total
L_total = L_sol + L_jup + L_sat

L_total_norm = np.linalg.norm(L_total)

print("El momento angular total del sistema es", L_total, "kg m^2 / s")
print("El momento angular total normalizado es", L_total_norm, "kg m^2 / s")
print("\n")
print("La norma del vector momento angular para el sol es", Lsol_norm, "kg m^2 / s")
print("La norma del vector momento angular para Júpiter es", Ljup_norm, "kg m^2 / s")
print("La norma del vector momento angular para Saturno es", Lsat_norm, "kg m^2 / s")

El momento angular total del sistema es [7.41746578e+35 2.15810379e+35 2.70674005e+37] kg m^2 / s
El momento angular total normalizado es 2.7078421843785097e+37 kg m^2 / s


La norma del vector momento angular para el sol es 3.030573241913306e+34 kg m^2 / s
La norma del vector momento angular para Júpiter es 1.9236938744506494e+37 kg m^2 / s
La norma del vector momento angular para Saturno es 7.812507134449024e+36 kg m^2 / s


**Conclusión:** El momento angular total del sistema corresponde casi por completo a la suma de los momentos angulares individuales del Sol, Júpiter y Saturno, dado que son los que aportan más contribución al cálculo por su masa. Esto indica que el comportamiento del sistema es el resultado de las contribuciones conjuntas de los momentos angulares de estos tres cuerpos principales.

Júpiter, con el mayor momento angular, domina la dinámica del sistema, seguido por Saturno. El Sol, en comparación, tiene un impacto menor debido a su baja velocidad de rotación y a que los momentos angulares de los planetas son mucho más grandes debido a sus distancias al Sol. Aunque Saturno está más alejado, su contribución al momento angular es menor que la de Júpiter, ya que su velocidad es inferior.

Por otro lado, el Sol actúa como un contrapeso a los planetas, dado que está relativamente cerca del baricentro del sistema solar.

Ahora, para encontrar la **inclinación del plano invariable de Laplace del sistema Sol-Júpiter-Saturno con respecto al plano de la eclíptica**, creamos un vector normal a la eclíptica, la inclinación esta dada por el ángulo formado entre el momento angular total y el eje z del vector normal, de la siguiente manera:

NOTA: El ángulo entre dos vectores se calcula así

 $$ \cos \theta = \frac{\vec{V_{1}} \cdot \vec{V_{2}}}{\|\vec{V_{1}}\| \|\vec{V_{2}}\|} $$

 donde $\vec{V_1}$ es el momento angular total del sistema y $\vec{V_2}$ es el vector normal a la eclíptica, $ (0,0,1) $.


In [6]:
ecl = np.array([0, 0, 1])
ecl_norm = np.linalg.norm(ecl)

# Se calcula el ángulo entre el vector normal de la eclíptica y L
i = np.arccos(np.dot(L_total, ecl) / (L_total_norm * ecl_norm))
i = np.degrees(i)

print("La inclinación del plano invariable de Laplace del sistema con respecto al plano de la eclíptica es", i, "°")

La inclinación del plano invariable de Laplace del sistema con respecto al plano de la eclíptica es 1.634778168001286 °


**Resultado:** La inclinación del plano invariable de Laplace del sistema con respecto al plano de la eclíptica es 1.635 °

Para **comparar la ubicación del plano invariable con la ubicación de los planos de las órbitas de Júpiter y Saturno** si se sabe que la inclinación de sus órbitas (de nuevo medidas respecto a la eclíptica) son respectivamente 1◦18′ y 2◦29′; se calcula la diferencia entre el angulo del plano invariable y el plano orbital de cada planeta.

In [7]:
# Inclinación de Júpiter
jup_inc_deg = 1 + 18/60  # 1 grado + 18 minutos / 60 minutos por grado
print("Inclinación de Júpiter en grados decimales:", jup_inc_deg, "°")

# Inclinación de Saturno
sat_inc_deg = 2 + 29/60  # 2 grados + 29 minutos / 60 minutos por grado
print("Inclinación de Saturno en grados decimales:", sat_inc_deg, "°")

Inclinación de Júpiter en grados decimales: 1.3 °
Inclinación de Saturno en grados decimales: 2.4833333333333334 °


In [8]:
# Diferencia con Júpiter
dif_jup = abs(i - jup_inc_deg)
print("Diferencia entre el plano invariable y la órbita de Júpiter:", dif_jup, "°")

# Diferencia con Saturno
dif_sat = abs(i - sat_inc_deg)
print("Diferencia entre el plano invariable y la órbita de Saturno:", dif_sat, "°")

Diferencia entre el plano invariable y la órbita de Júpiter: 0.334778168001286 °
Diferencia entre el plano invariable y la órbita de Saturno: 0.8485551653320473 °


**Conclusión:** El plano invarible tiene menos inclinación con respecto a la órbita de Júpiter que con la órbita de Saturno, esto se debe a que el cuerpo más masivo es el que indica la dirección del plano invariante en un sistema, en este caso del sistema solar (ya que usamos los cuerpos más pesados y las masas de los demás planetas se podrían despreciar).

## Resultados 100 años después

In [9]:
# Vectores de estado del Sol, Júpiter, Saturno

pos_sol100 = np.array([1.149498606342612E+06, 2.927257075792357E+05, -3.168656612444807E+0]) * 1000 # en m
vel_sol100 = np.array([-5.795966467487191E-03, 1.150548260551569E-02, 1.998520726376619E-05]) * 1000 # en m/s

pos_jup100 = np.array([-6.205253881180364E+08, -5.223129731549698E+08, 1.604870896421686E+07]) * 1000 # en m
vel_jup100 = np.array([8.244055791548094E+00, -9.383969280875352E+00, -1.442666306613680E-01]) * 1000 # en m/s

pos_sat100 = np.array([-1.063299599343205E+09, 8.680977385481120E+08, 2.757619635067564E+07]) * 1000 # en m
vel_sat100 = np.array([-6.625840340425324E+00, -7.507210506293189E+00, 3.939152509718982E-01]) * 1000 # en m/s

In [10]:
# Momento angular
L_sol100 = np.cross(pos_sol100, vel_sol100) * m_sol
L_jup100 = np.cross(pos_jup100, vel_jup100) * m_jup
L_sat100 = np.cross(pos_sat100, vel_sat100) * m_sat

# Normalizar momento angular (magnitud escalar)
Lsol100_norm = np.linalg.norm(L_sol100)
Ljup100_norm = np.linalg.norm(L_jup100)
Lsat100_norm = np.linalg.norm(L_sat100)

# Momento angular total
L_total100 = L_sol100 + L_jup100 + L_sat100

L_total_norm100 = np.linalg.norm(L_total100)

print("El momento angular total del sistema es", L_total100, "kg m^2 / s")
print("El momento angular total normalizado es", L_total_norm100, "kg m^2 / s")
print("\n")
print("El momento angular del sol es", L_sol100, "kg m^2 / s")
print("El momento angular de Júpiter es", L_jup100, "kg m^2 / s")
print("El momento angular de Saturno es", L_sat100, "kg m^2 / s")
print("La norma del vector momento angular para el sol es", Lsol100_norm, "kg m^2 / s")
print("La norma del vector momento angular para Júpiter es", Ljup100_norm, "kg m^2 / s")
print("La norma del vector momento angular para Saturno es", Lsat100_norm, "kg m^2 / s")

El momento angular total del sistema es [7.41057776e+35 2.15415585e+35 2.70670631e+37] kg m^2 / s
El momento angular total normalizado es 2.7078062626082524e+37 kg m^2 / s


El momento angular del sol es [ 1.17050548e+31 -4.56431581e+31  2.96713794e+34] kg m^2 / s
El momento angular de Júpiter es [4.28975620e+35 8.12287337e+34 1.92300256e+37] kg m^2 / s
El momento angular de Saturno es [3.12070451e+35 1.34232495e+35 7.80736610e+36] kg m^2 / s
La norma del vector momento angular para el sol es 2.967141679854324e+34 kg m^2 / s
La norma del vector momento angular para Júpiter es 1.9234981247949903e+37 kg m^2 / s
La norma del vector momento angular para Saturno es 7.814753469686474e+36 kg m^2 / s


In [11]:
ecl = np.array([0, 0, 1])
ecl_norm = np.linalg.norm(ecl)

# Se calcula el ángulo entre el vector normal de la eclíptica y L
a = np.arccos(np.dot(L_total100, ecl) / (L_total_norm100 * ecl_norm))
a = np.degrees(a)

print("La inclinación del plano invariable de Laplace del sistema con respecto al plano de la eclíptica 100 años después es", a, "°")

La inclinación del plano invariable de Laplace del sistema con respecto al plano de la eclíptica 100 años después es 1.633166432834246 °


In [12]:
# Diferencia de ángulos en la inclinación del plano invariable del sistema y la eclíptica
e = a - i
e

-0.0016117351670401625

In [13]:
# Diferencia con Júpiter
dif_jup100 = abs(a - jup_inc_deg)
print("Diferencia entre el plano invariable y la órbita de Júpiter 100 años después:", dif_jup100, "°")

# Diferencia con Saturno
dif_sat100 = abs(a - sat_inc_deg)
print("Diferencia entre el plano invariable y la órbita de Saturno 100 años después:", dif_sat100, "°")

Diferencia entre el plano invariable y la órbita de Júpiter 100 años después: 0.33316643283424585 °
Diferencia entre el plano invariable y la órbita de Saturno 100 años después: 0.8501669004990875 °


In [14]:
# Diferencia entre los angulos del plano invariante y los planos orbitales
j = dif_jup100 - dif_jup
s = dif_sat100 - dif_sat

j,s

(-0.0016117351670401625, 0.0016117351670401625)

Comparardo los resultados de hoy y luego de 100 años tenemos que los ordenes de magnitud para el **momento angular** son iguales, los cambios en los decimales son mínimos, de esto podemos concluir que el momento angular es una constante en los sistemas de n-cuerpos.

Además, La **inclinación del plano invariable de Laplace** del sistema con respecto al plano de la eclíptica 100 años después es similar a la que nos da hoy, con un valor de 1.633°, tiene una inclinación **menor de 0.002°**

Y por último, para el ángulo entre el plano invariable y la órbita de **Júpiter** hay un **reducción en el ángulo de 0.002°** y para **Saturno** hay un **incremento en el ángulo de 0.002°**

### **2. Plano invariable del sistema Sol-Júpiter-Saturno (Parte 2)**

Repita el punto anterior utilizando las herramientas computacionales vistas en clase con el profesor Jorge Zuluaga.

In [15]:
!pip install -Uq pymcel

   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 1.2/1.2 MB 7.4 MB/s eta 0:00:00
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 5.3/5.3 MB 34.2 MB/s eta 0:00:00
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 1.7/1.7 MB 21.2 MB/s eta 0:00:00
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 910.8/910.8 kB 18.5 MB/s eta 0:00:00


In [16]:
import pymcel as pc

Paquete pymcel cargado. Versión: 0.6.10


In [17]:
pc.descarga_kernels()

In [18]:
import spiceypy as spy

In [19]:
spy.furnsh([
    'pymcel/data/gm_de431.tpc',
    'pymcel/data/de430.bsp',
    'pymcel/data/latest_leapseconds.tls'
])

Se crea el diccionario *planet_datos* para almacenar los datos de los planetas y se crea otro diccionario *mu_planeta* para almacenar los μ de cada planeta, siendo $μ = GM$, luego se consulta con `pc.consulta_horizonts` los vectores de estado (vector posición $\vec{r}$ y el vector velocidad $\vec{v}$) de cada uno para almacenarlos en el diccionario *planet_datos* para poder consultar cada vector correspondiente al id de su planeta; y por último con `SPICE` se obtienen los μ y se almacenan en el diccionario *mu_planetas*.

In [20]:
planet_datos = {}
mu_planeta = {}

for id in ['Sun',
           'Jupiter Barycenter',
           'Saturn Barycenter']:

  tabla, df, datos = pc.consulta_horizons(
    id=id,
    location='@0',
    epochs='2024-10-31 00:00:00',
    datos='vectors'
  )
  mu_cuerpo = spy.bodvrd(id,'GM',1)[1][0]

  mu_planeta[id] = mu_cuerpo * 1000 # cambio de km^3 / s^2 a m^3 / s^2
  planet_datos[id] = datos

In [21]:
planet_datos # vector posicion y velocidad de cada id [m y m/s]

{'Sun': array([-9.22313414e+08, -6.97157771e+08,  2.78942338e+07,  1.18320298e+01,
        -7.57637262e+00, -1.81751999e-01]),
 'Jupiter Barycenter': array([ 2.25609572e+11,  7.21961463e+11, -8.04224247e+09, -1.26188359e+04,
         4.51964012e+03,  2.63598069e+02]),
 'Saturn Barycenter': array([ 1.40695646e+12, -3.15292334e+11, -5.05358341e+10,  1.57505794e+03,
         9.40596140e+03, -2.26272274e+02])}

In [22]:
mu_planeta # mu de cada id [m^3 / s^2]

{'Sun': 132712440041939.3,
 'Jupiter Barycenter': 126712764800.0002,
 'Saturn Barycenter': 37940585200.0}

Se cambian los diccionarios que contienen los vectores de estado a arrays de `numpy`, luego se almacenan las posiciones de cada planeta en la variable Rs y las velocidades de cada planeta en Vs

In [23]:
Rs = np.array([planet_datos[id][0:3] for id in planet_datos])
Vs = np.array([planet_datos[id][3:6] for id in planet_datos])

In [24]:
Rs # vector posicion de cada id [m]

array([[-9.22313414e+08, -6.97157771e+08,  2.78942338e+07],
       [ 2.25609572e+11,  7.21961463e+11, -8.04224247e+09],
       [ 1.40695646e+12, -3.15292334e+11, -5.05358341e+10]])

In [25]:
Vs # vector velocidad de cada id [m/s^2]

array([[ 1.18320298e+01, -7.57637262e+00, -1.81751999e-01],
       [-1.26188359e+04,  4.51964012e+03,  2.63598069e+02],
       [ 1.57505794e+03,  9.40596140e+03, -2.26272274e+02]])

Se hace una lista por comprensión para almacenar las masas de cada planeta, lo que corresponde a dividir cada μ por la constante gravitacional $G$

In [26]:
masas_planetas = np.array([mu_planeta[id] for id in planet_datos])
masas_planetas = masas_planetas / pc.constantes.G
masas_planetas

array([1.98840987e+24, 1.89851767e+21, 5.68457894e+20])

Se almacena en la variable *L_planetas* el momento angular de cada cuerpo, el cual se define así:

$$\vec{L} = ∑_i m_i (\vec{r}_i × \vec{v}_i)$$

donde $m$ son las masas de cada cuerpo, $\vec{r}$ son los vectores posición y $\vec{v}$ son los vectores velocidad.

In [27]:
L_planetas = np.cross(Rs, Vs) * masas_planetas[:,None]

print("El momento angular para el sol es", L_planetas[0], "kg m^2 / s")
print("El momento angular para Júpiter es", L_planetas[1], "kg m^2 / s")
print("El momento angular para Saturno es", L_planetas[2], "kg m^2 / s")

El momento angular para el sol es [6.72175848e+32 3.22943810e+32 3.02965694e+34] kg m^2 / s
El momento angular para Júpiter es [4.30309830e+35 7.97633546e+34 1.92319599e+37] kg m^2 / s
El momento angular para Saturno es [3.10764572e+35 1.35724080e+35 7.80514395e+36] kg m^2 / s


Se almacena el momento angular total del sistema solar en la variable *L_total* sumando componente a componente del momento angular de cada cuerpo.

In [28]:
L_total2 = [sum(L_planetas[:, i]) for i in range(3)]
L_total2 = np.array(L_total2)
print("Momentum angular total del Sistema Sol-Júpiter-Saturno:", L_total2)

Momentum angular total del Sistema Sol-Júpiter-Saturno: [7.41746578e+35 2.15810379e+35 2.70674004e+37]


Ahora, se calcula la norma del momento angular de cada cuerpo y se almacena en *l_norma_planeta*.

In [29]:
l_norma_planeta = []

for planeta in L_planetas:
  norma = np.linalg.norm(planeta)

  l_norma_planeta.append(norma)

In [30]:
print("El momento angular normalizado del sol es", l_norma_planeta[0], "kg m^2 / s")
print("El momento angular normalizado de Júpiter es", l_norma_planeta[1], "kg m^2 / s")
print("El momento angular normalizado de Saturno es", l_norma_planeta[2], "kg m^2 / s")

El momento angular normalizado del sol es 3.0305745783424192e+34 kg m^2 / s
El momento angular normalizado de Júpiter es 1.923693871817711e+37 kg m^2 / s
El momento angular normalizado de Saturno es 7.812507131244726e+36 kg m^2 / s


In [31]:
l_norma = np.linalg.norm(L_total2)
print("El momento angular total normalizado es", l_norma, "kg m^2 / s")

El momento angular total normalizado es 2.7078421827616205e+37 kg m^2 / s


**Conclusión:** Podemos concluir que los resultados por este método son iguales a hacer el cálculo manual, la librería de `pymcel` nos facilita la extracción de datos al programar.

Júpiter, con el mayor momento angular, domina la dinámica del sistema, seguido por Saturno. El Sol, en comparación, tiene un impacto menor debido a su baja velocidad de rotación y a que los momentos angulares de los planetas son mucho más grandes debido a sus distancias al Sol. Aunque Saturno está más alejado, su contribución al momento angular es menor que la de Júpiter, ya que su velocidad es inferior.

Por otro lado, el Sol actúa como un contrapeso a los planetas, dado que está relativamente cerca del baricentro del sistema solar.

Ahora, para encontrar **la inclinación del plano invariable de Laplace del sistema Sol-Júpiter-Saturno con respecto al plano de la eclíptica**, creamos un vector normal a la eclíptica, la inclinación esta dada por el ángulo formado entre el momento angular total y el eje z del vector norma, este vector es $(0,0,1)$, de esta manera:

In [32]:
ecl2 = np.array([0, 0, 1])
ecl2_norm = np.linalg.norm(ecl2)

# Se calcula el ángulo entre el vector normal de la eclíptica y L
i2 = np.arccos(np.dot(L_total2, ecl2) / (l_norma * ecl2_norm))
i2 = np.degrees(i2)

print("La inclinación del plano invariable de Laplace del sistema con respecto al plano de la eclíptica es", i2, "°")

La inclinación del plano invariable de Laplace del sistema con respecto al plano de la eclíptica es 1.6347781678741904 °


**Resultado:** La inclinación del plano invariable de Laplace del sistema con respecto al plano de la eclíptica es 1.635 °

Para **comparar la ubicación del plano invariable con la ubicación de los planos de las órbitas de Júpiter y Saturno** si se sabe que la inclinación de sus órbitas (de nuevo medidas respecto a la eclíptica) son respectivamente 1◦18′ y 2◦29′; se calcula la diferencia entre el angulo del plano invariable y el plano orbital de cada planeta.

In [33]:
# Diferencia con Júpiter
dif_jup2 = abs(i2 - jup_inc_deg)
print("Diferencia entre el plano invariable y la órbita de Júpiter:", dif_jup2, "°")

# Diferencia con Saturno
dif_sat2 = abs(i2 - sat_inc_deg)
print("Diferencia entre el plano invariable y la órbita de Saturno:", dif_sat2, "°")

Diferencia entre el plano invariable y la órbita de Júpiter: 0.33477816787419035 °
Diferencia entre el plano invariable y la órbita de Saturno: 0.848555165459143 °


**Conclusión:** El plano invarible tiene menos inclinación con respecto a la órbita de Júpiter que con la órbita de Saturno, esto se debe a que el cuerpo más masivo es el que indica la dirección del plano invariante en un sistema, en este caso del sistema solar (ya que usamos los cuerpos más pesados y las masas de los demás planetas se podrían despreciar).

## Resultados 100 años después

In [34]:
planet_datos2 = {}
mu_planeta2 = {}

for id in ['Sun',
           'Jupiter Barycenter',
           'Saturn Barycenter']:

  tabla, df, datos = pc.consulta_horizons(
    id=id,
    location='@0',
    epochs='2124-10-31 00:00:00',
    datos='vectors'
  )
  mu_cuerpo2 = spy.bodvrd(id,'GM',1)[1][0]

  mu_planeta2[id] = mu_cuerpo2 * 1000 # cambio de km^3 / s^2 a m^3 / s^2
  planet_datos2[id] = datos

/usr/local/lib/python3.10/dist-packages/erfa/core.py:133: ErfaWarning: ERFA function "dtf2d" yielded 1 of "dubious year (Note 6)"
  warn(f'ERFA function "{func_name}" yielded {wmsg}', ErfaWarning)


In [35]:
Rs_2 = np.array([planet_datos2[id][0:3] for id in planet_datos2])
Vs_2 = np.array([planet_datos2[id][3:6] for id in planet_datos2])

In [36]:
masas_planetas = np.array([mu_planeta2[id] for id in planet_datos2])
masas_planetas = masas_planetas / pc.constantes.G
masas_planetas

array([1.98840987e+24, 1.89851767e+21, 5.68457894e+20])

In [37]:
# Momento angular de los cuerpos 100 años después
L_planetas2 = np.cross(Rs_2, Vs_2) * masas_planetas[:,None]

# Normalización del momento angular total
Ltotal2 = [sum(L_planetas2[:, i]) for i in range(3)]
Ltotal2 = np.array(Ltotal2)

In [38]:
# Normalización de cada momento angular
l_norma_planeta2 = []

for planeta in L_planetas2:
  norma = np.linalg.norm(planeta)

  l_norma_planeta2.append(norma)

In [39]:
l_norma2 = np.linalg.norm(Ltotal2)

In [40]:
print("El momento angular para el sol es", L_planetas2[0], "kg m^2 / s")
print("El momento angular para Júpiter es", L_planetas2[1], "kg m^2 / s")
print("El momento angular para Saturno es", L_planetas2[2], "kg m^2 / s")
print("El momento angular total del sistema es", Ltotal2, "kg m^2 / s")
print("\n")
print("El momento angular normalizado del sol es", l_norma_planeta2[0], "kg m^2 / s")
print("El momento angular normalizado de Júpiter es", l_norma_planeta2[1], "kg m^2 / s")
print("El momento angular normalizado de Saturno es", l_norma_planeta2[2], "kg m^2 / s")
print("El momento angular total normalizado es", l_norma2, "kg m^2 / s")

El momento angular para el sol es [7.36545807e+32 3.19500370e+32 2.96713865e+34] kg m^2 / s
El momento angular para Júpiter es [4.28975620e+35 8.12287337e+34 1.92300256e+37] kg m^2 / s
El momento angular para Saturno es [3.12070451e+35 1.34232495e+35 7.80736610e+36] kg m^2 / s
El momento angular total del sistema es [7.41782616e+35 2.15780729e+35 2.70670631e+37] kg m^2 / s


El momento angular normalizado del sol es 2.9682246509567994e+34 kg m^2 / s
El momento angular normalizado de Júpiter es 1.923498122874928e+37 kg m^2 / s
El momento angular normalizado de Saturno es 7.81475346665893e+36 kg m^2 / s
El momento angular total normalizado es 2.707808536503223e+37 kg m^2 / s


In [41]:
ecl2 = np.array([0, 0, 1])
ecl2_norm = np.linalg.norm(ecl2)

# Se calcula el ángulo entre el vector normal de la eclíptica y L
i_2 = np.arccos(np.dot(Ltotal2, ecl2) / (l_norma2 * ecl2_norm))
i_2 = np.degrees(i_2)

print("La inclinación del plano invariable de Laplace del sistema con respecto al plano de la eclíptica es", i_2, "°")

La inclinación del plano invariable de Laplace del sistema con respecto al plano de la eclíptica es 1.634854203894085 °


In [42]:
# Diferencia de ángulos en la inclinación del plano invariable del sistema y la eclíptica
e2 = i_2 - i2
e2

7.603601989458753e-05

In [43]:
# Diferencia con Júpiter
dif_jup2_100 = abs(i_2 - jup_inc_deg)
print("Diferencia entre el plano invariable y la órbita de Júpiter:", dif_jup2_100, "°")

# Diferencia con Saturno
dif_sat2_100 = abs(i_2 - sat_inc_deg)
print("Diferencia entre el plano invariable y la órbita de Saturno:", dif_sat2_100, "°")

Diferencia entre el plano invariable y la órbita de Júpiter: 0.33485420389408493 °
Diferencia entre el plano invariable y la órbita de Saturno: 0.8484791294392484 °


In [44]:
# Diferencia entre los angulos del plano invariante y los planos orbitales
j2 = dif_jup2_100 - dif_jup2
s2 = dif_sat2_100 - dif_sat2

j2 ,s2

(7.603601989458753e-05, -7.603601989458753e-05)

Comparardo los resultados de hoy y luego de 100 años por este segundo método tenemos que los ordenes de magnitud para el **momento angular** son iguales, los cambios en los decimales son mínimos, de esto podemos concluir que el momento angular es una constante en los sistemas de n-cuerpos.

Además, La **inclinación del plano invariable de Laplace** del sistema con respecto al plano de la eclíptica 100 años después es similar a la que nos da hoy, tiene una inclinación **menor de 0.00076°**

Y por último, para el ángulo entre el plano invariable y la órbita de **Júpiter** hay un **reducción en el ángulo de 0.00076°** y para **Saturno** hay un **incremento en el ángulo de 0.00076°**

Por este método se tuvo más exactitud en la variación del plano invariable de Lapalace y de este con respecto a los planos orbitales de Júpiter y Saturno.